setcookie: Add note regarding SameSite=None and disabled Secure behavior

AllenJB opened the pull request:
Fixes #4873

AllenJB committed:
setcookie: Add note regarding SameSite=None and disabled Secure behavior

TimWolla reviewed (approved):
@Girgias might have markup opinions here?

Girgias commented on reference/network/functions/setcookie.xml:
```diff
@@ -178,6 +178,15 @@
+         blocked by the client.
+        </para>
+       </note>
+      </para>
```
```suggestion
      <note>
       <simpara>
        To set a cookie that includes attributes that aren't among the keys listed,
        use <function>header</function>.
       </simpara>
      </note>

      <note>
       <simpara>
        If <literal>samesite</literal> is <literal>None</literal> then
        <literal>secure</literal> must also be enabled or the cookie will be
        blocked by the client.
       </simpara>
      </note>
```

Girgias reviewed (changes requested):
Indeed I have some markup notes, the wrapping para is very much unnecessary and please use `<simpara>` and fix the above note at the same time :)

AllenJB committed:
setcookie (review amends - reviewed markup across this page)

Girgias commented on reference/network/functions/setcookie.xml:
```diff
@@ -80,23 +80,21 @@
+        You may notice the <parameter>expires_or_options</parameter> parameter takes on a
+        Unix timestamp, as opposed to the date format <literal>Wdy, DD-Mon-YYYY
+        HH:MM:SS GMT</literal>, this is because PHP does this conversion
+        internally.
```
While at it could you rewrite this to get rid of the personalization (i.e. using "you") in accordance with the manual style guide.

Girgias commented on reference/network/functions/setcookie.xml:
```diff
@@ -245,22 +247,20 @@ setcookie("TestCookie", $value, time()+3600, "/~rasmus/", "example.com", true);
+  </simpara>
+  <informalexample>
+   <programlisting role="php">
+    <![CDATA[
```
```suggestion
   <programlisting role="php">
<![CDATA[
```

Girgias commented on reference/network/functions/setcookie.xml:
```diff
@@ -319,38 +315,37 @@ if (isset($_COOKIE['cookie'])) {
-   <para>
+   <simpara>
     You can use output buffering to send output prior to the
     call of this function, with the overhead of all of your output to the
```
Personalization

Girgias commented on reference/network/functions/setcookie.xml:
```diff
@@ -186,55 +189,54 @@
-    <title><function>setcookie</function> send example</title>
-    <programlisting role="php">
-     <![CDATA[
+  </simpara>
```
Is this paragraph even useful?

Girgias commented on reference/network/functions/setcookie.xml:
```diff
@@ -186,55 +189,54 @@
+  <example>
+   <title><function>setcookie</function> send example</title>
+   <programlisting role="php">
+    <![CDATA[
```
And quite a few other places do this
```suggestion
<![CDATA[
```

Girgias reviewed (changes requested):
Sorry for the delay, but going throught the file just brings to light more issues :/

AllenJB committed:
setcookie (review amends - CDATA indentation; Personalization)

AllenJB committed:
setcookie (review amends - trailing whitespace)

AllenJB commented:
I've fixed all the review issues, including CDATA indents and trailing whitespace.

Additionally, I changed the suggestion to consider explode() for setting multiple values in a single cookie to json_encode() as I think this is easier / less error prone to use correctly.

Girgias reviewed (approved):
Thank you!
